crates/context/src/lib.rs

```diff
@@ -1021,4 +1023,77 @@ impl ContextManager {
 
         Ok(Some(stream))
     }
+
+    pub fn is_application_blob_installed(&self, blob_id: BlobId) -> EyreResult<bool> {
+        self.blob_manager.has(blob_id)
+    }
+
+    pub async fn propose(
+        &self,
+        context_id: ContextId,
+        signer_id: PublicKey,
+        proposal_id: ProposalId,
+        actions: Vec<ProposalAction>,
+    ) -> EyreResult<()> {
+        let handle = self.store.handle();
+        let Some(context_config) = handle.get(&ContextConfigKey::new(context_id))? else {
+            bail!(
+                "Failed to retrieve ContextConfig for context ID: {}",
+                context_id
+            );
+        };
+
+        let Some(ContextIdentityValue {
+            private_key: Some(signing_key),
+        }) = handle.get(&ContextIdentityKey::new(context_id, signer_id))?
+        else {
+            bail!("No private key found for signer");
+        };
+
+        let _ = self
+            .config_client
+            .mutate::<ContextProxy>(
+                context_config.protocol.as_ref().into(),
+                context_config.network.as_ref().into(),
+                context_config.contract.as_ref().into(),
+            )
+            .propose(proposal_id, signer_id.rt().unwrap(), actions)
```

Comment:
```suggestion
            .propose(proposal_id, signer_id.rt().expect("infallible conversion"), actions)
```

while we want to avoid panic points, any thing that is "safe" to use unwrap, must define why that's the case